web-app/src/main/webapp/script.js

```diff
@@ -70,17 +75,64 @@ function showReportForm() {
 }
 
 async function loadPoliceReports(map) {
+  // Clear all markers on the map
+  for (const marker of mapMarkers) {
+    marker.setMap(null);
+  }
+  mapMarkers.length = 0;
```

Comment:
nit: let's do `mapMarkets = []` since it's more obvious what it's doing? Changing the `length` seems subtle.